Solutions/Salesforce Service Cloud/Analytic Rules/Salesforce-BruteForce.yaml

```diff
@@ -0,0 +1,42 @@
+id: 5a6ce089-e756-40fb-b022-c8e8864a973a
+name: Brute force attack against user credentials
+description: |
+  'Identifies evidence of brute force activity against a user based on multiple authentication failures 
+  and at least one successful authentication within a given time window. Note that the query does not enforce any sequence,
+  and does not require the successful authentication to occur last.
+  The default failure threshold is 10, success threshold is 1, and the default time window is 20 minutes.'
+severity: Medium
+requiredDataConnectors:
+  - connectorId: SalesforceServiceCloud
+    dataTypes:
+      - SalesforceServiceCloud
+queryFrequency: 20m
+queryPeriod: 20m
+triggerOperator: gt
+triggerThreshold: 0
+tactics:
+  - CredentialAccess
+relevantTechniques:
+  - T1110
+query: |
+  let failureCountThreshold = 10;
+  let successCountThreshold = 1;
+  SalesforceServiceCloud
+  | where EventType == "Login"
+  | summarize 
+        StartTime = min(TimeGenerated), 
+        EndTime = max(TimeGenerated), 
+        IpAddresses = make_set (ClientIp, 100),
+        FailureCount = countif(LoginStatus !='LOGIN_NO_ERROR'),
+        SuccessCount = countif(LoginStatus =='LOGIN_NO_ERROR')
+    by User, UserId, UserType 
+    | where FailureCount >= failureCountThreshold and SuccessCount >= successCountThreshold
+    | extend
+        IpAddresses = strcat_array(IpAddresses, ", ")
```

Comment:
Why are we doing this, this is already an array?

Reply:
Fixed